model/group_syncable.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type GroupSyncableType int
+
+const (
+	GroupSyncableTypeTeam GroupSyncableType = iota
+	GroupSyncableTypeChannel
+)
```

Comment:
No particular reason I suppose, it could just as well be this:

diff --git a/model/group_syncable.go b/model/group_syncable.go
index 3e1535e0b..d529ca513 100644
--- a/model/group_syncable.go
+++ b/model/group_syncable.go
@@ -10,16 +10,15 @@ import (
 	"net/http"
 )
 
-type GroupSyncableType int
+type GroupSyncableType string
 
 const (
-	GroupSyncableTypeTeam GroupSyncableType = iota
-	GroupSyncableTypeChannel
+	GroupSyncableTypeTeam    GroupSyncableType = "Team"
+	GroupSyncableTypeChannel GroupSyncableType = "Channel"
 )
 
 func (gst GroupSyncableType) String() string {
-	// Order matters. Keep in sync with iotas.
-	return [...]string{"Team", "Channel"}[gst]
+	return string(gst)
 }
 
 type GroupSyncable struct {

Reply:
Thoughts? I'm 0/5.

Reply:
I think I'd prefer the proposed change as it's more consistent with what we do elsewhere, and just feels a bit less confusing having only one "value" for those constants.